From f61e650ca55b6466b70a2e0870b053a39e428c6e Mon Sep 17 00:00:00 2001
From: Michal Simek <monstr@monstr.eu>
Date: Mon, 16 Apr 2012 08:57:22 +0200
Subject: [PATCH] remoteproc: Support firmware module parameter

Passing firmware module parameter can override default
firmware name written in DTS.

Signed-off-by: Michal Simek <monstr@monstr.eu>
---
 drivers/remoteproc/zynq_remoteproc.c |   13 ++++++++++++-
 1 files changed, 12 insertions(+), 1 deletions(-)

diff --git a/drivers/remoteproc/zynq_remoteproc.c b/drivers/remoteproc/zynq_remoteproc.c
index b3994a7..4ddd69a 100644
--- a/drivers/remoteproc/zynq_remoteproc.c
+++ b/drivers/remoteproc/zynq_remoteproc.c
@@ -36,6 +36,9 @@
 
 #include "remoteproc_internal.h"
 
+/* Module parameter */
+static char *firmware;
+
 /* Structure for storing IRQs */
 struct irq_list {
 	int irq;
@@ -238,7 +241,12 @@ static int __devinit zynq_remoteproc_probe(struct platform_device *pdev)
 		goto ipi_fault;
 	}
 
-	prop = of_get_property(pdev->dev.of_node, "firmware", NULL);
+	/* Module param firmware first */
+	if (firmware)
+		prop = firmware;
+	else
+		prop = of_get_property(pdev->dev.of_node, "firmware", NULL);
+
 	if (prop) {
 		dev_dbg(&pdev->dev, "Using firmware: %s\n", prop);
 		local->rproc = rproc_alloc(&pdev->dev, dev_name(&pdev->dev),
@@ -301,5 +309,8 @@ static struct platform_driver zynq_remoteproc_driver = {
 };
 module_platform_driver(zynq_remoteproc_driver);
 
+module_param(firmware, charp, 0);
+MODULE_PARM_DESC(firmware, "Override the firmware image name. Default value in DTS.");
+
 MODULE_LICENSE("GPL v2");
 MODULE_DESCRIPTION("Zynq remote processor control driver");
-- 
1.7.0.4

